src/Configuration.php

```diff
@@ -593,6 +595,20 @@ public function setSchemaIgnoreClasses(array $schemaIgnoreClasses): void
         $this->attributes['schemaIgnoreClasses'] = $schemaIgnoreClasses;
     }
 
+    public function isLazyProxyEnabled(): bool
+    {
+        return $this->attributes['lazyProxy'] ?? false;
+    }
+
+    public function setLazyProxyEnabled(bool $lazyProxy): void
+    {
+        if (PHP_VERSION_ID < 80400) {
+            throw new LogicException('Lazy loading proxies require PHP 8.4 or higher.');
+        }
+
+        $this->attributes['lazyProxy'] = $lazyProxy;
+    }
+
```

Comment:
I suggest to use the term "native lazy object", as there's no longer any notion of "proxy" with this implementation, and it's a **native** feature of the language.

```suggestion
    public function isNativeLazyObjectEnabled(): bool
    {
        return $this->attributes['nativeLazyObject'] ?? false;
    }

    public function setNativeLazyObjectEnabled(bool $nativeLazyObject): void
    {
        if (PHP_VERSION_ID < 80400) {
            throw new LogicException('Native lazy object require PHP 8.4 or higher.');
        }

        $this->attributes['nativeLazyObject'] = $nativeLazyObject;
    }

```